lib/src/iterable_matchers.dart

```diff
@@ -155,45 +157,53 @@ class _UnorderedMatches extends Matcher {
       : _expected = expected.map(wrapMatcher).toList();
 
   String _test(item) {
-    if (item is Iterable) {
-      var list = item.toList();
+    if (item is! Iterable) return 'not iterable';
 
-      // Check the lengths are the same.
-      if (_expected.length > list.length) {
-        return 'has too few elements (${list.length} < ${_expected.length})';
-      } else if (_expected.length < list.length) {
-        return 'has too many elements (${list.length} > ${_expected.length})';
-      }
+    var values = item.toList();
 
-      var matched = new List<bool>.filled(list.length, false);
-      var expectedPosition = 0;
-      for (var expectedMatcher in _expected) {
-        var actualPosition = 0;
-        var gotMatch = false;
-        for (var actualElement in list) {
-          if (!matched[actualPosition]) {
-            if (expectedMatcher.matches(actualElement, {})) {
-              matched[actualPosition] = gotMatch = true;
-              break;
-            }
-          }
-          ++actualPosition;
-        }
+    // Check the lengths are the same.
+    if (_expected.length > values.length) {
+      return 'has too few elements (${values.length} < ${_expected.length})';
+    } else if (_expected.length < values.length) {
+      return 'has too many elements (${values.length} > ${_expected.length})';
+    }
 
-        if (!gotMatch) {
-          return new StringDescription()
-              .add('has no match for ')
-              .addDescriptionOf(expectedMatcher)
-              .add(' at index $expectedPosition')
-              .toString();
+    var edges = values.map((_) => <int>[]).toList(growable: false);
```

Comment:
nit: you could use `List.filled`

Reply:
I don't think I can because that would give us the same instance of a list at every index.

Reply:
Sorry meant `List.generate`

Reply:
Sure, it's a few extra characters for now but once we can remove `new` it'll fit on one line again and I think the intent is more clear that the `.toList()` call.